packages/composites/proof-of-reserves/src/endpoint/reserves.ts

```diff
@@ -4,39 +4,55 @@ import { makeOptions } from '../config'
 import { Indexer, runBalanceAdapter } from '../utils/balance'
 import { runProtocolAdapter } from '../utils/protocol'
 import { runReduceAdapter } from '../utils/reduce'
-
+import { getValidAddresses } from '../utils/addressValidator'
 export const supportedEndpoints = ['reserves']
 
-const paramOptions = makeOptions()
+export const execute: ExecuteWithConfig<Config> = async (input, context, config) => {
+  const paramOptions = makeOptions()
 
-const inputParameters: InputParameters = {
-  protocol: {
-    required: true,
-    type: 'string',
-    description: 'The protocol external adapter to use',
-    options: paramOptions.protocol,
-  },
-  indexer: {
-    required: true,
-    type: 'string',
-    description: 'The indexer external adapter to use',
-    options: paramOptions.indexer,
-  },
-  confirmations: {
-    required: false,
-    type: 'number',
-    description:
-      'The number of confirmations required for a transaction to be counted when getting an address balance',
-    default: 6,
-  },
-  addresses: {
-    required: false,
-    type: 'array',
-    description: 'An array of addresses to get the balance from, when `protocol` is set to `list`',
-  },
-}
+  const inputParameters: InputParameters = {
```

Comment:
Why is this moving within `execute`'s function scope? It means that it can no longer get exported and utilized by the EA framework for things such as cache key generation.

Reply:
This resolves that issue I had with not being able to set env vars for testing.  I can elaborate further in a huddle, but this is necessary.

Reply:
The problem is, `makeOptions`, which gets the env vars, runs BEFORE the first line of `adapter.test.ts` if it is left outside of the `execute` function.  So by the time I try to set env vars, the options have already been created so I can't change them.  This is because `makeOptions` is executed when Jest imports all the required deps to run the tests, which is BEFORE any of the tests actually execute.

Reply:
also, the options object is not exported, so it shouldn't have any external impact where the function is located.

Reply:
I agree with Justin's points, we do need the input params to be exported before and kept consistent with other EAs. For the particular issue of testing it, you can use Jest to mock and shadow objects in the global scope.

Reply:
+1 to that this should be resolved within the test and the endpoint maintained